tools/integration_tests_runner.py

```diff
@@ -1,137 +1,146 @@
-            try:
-                test_file, test = self.tests.get()
-                self.exec_test(test_file, test, clean_env=(test_file in TEMPFOLDER_TESTS))
 
-            except:
-                logging.error(traceback.format_exc())
-                return
+def exec_test(a_tuple):
+    """
+    Executes a test and returns the result
+    :param a_tuple: a tuple consisting of
+    * the path to the test file (e.g. ..../base_tests)
+    * the name of the test to be executed from that file
+    * dictionary with all the parameters to be passed to the executable. At
+    this point the keys may contain --user_resource_path and --data_path if the
+    executable file is not in the list of the tests that require temporary folders
 
+    :return: (the name of the file, the name of the test),
+    (standard error from the test, test's exit code)
+    """
+    test_file, test, params = a_tuple
+    params[FILTER_KEY] = test
 
-    def log_start_finish(self, test_file, keys, start=False, finish=False):
-        if not self.write_start_finish_log:
-            return
+    out, err, result = None, None, None
+    try:
+        if test_file in TEMPFOLDER_TESTS:
+            tmpdir = tempfile.mkdtemp()
+            params[DATA_PATH_KEY] = tmpdir
 
-        if (not start and not finish) or (start and finish):
-            logging.warning("You need to pass either start=True or finish=True, but only one of them! You passed start={0}, finish={1}".format(start, finish))
-            return
+        keys = params_from_dict(params)
+        out, err, result = execute_file(test_file, keys)
 
-        string = "Started" if start else "Finished"
+        if test_file in TEMPFOLDER_TESTS:
+            shutil.rmtree(tmpdir)
 
-        with self.start_finish_lock:
-            self.start_finish_log.write("{string} {test_file} {keys} at {time}\n".format(string=string, test_file=test_file, keys=keys, time=time()))
-            self.start_finish_log.flush()
+    except:
+        for tup in traceback.extract_stack():
+            get_logger().error(tup)
 
+    finally:
+        return (test_file, test), (err, result)
 
-    def exec_test(self, test_file, test, clean_env=False):
-        keys = '"--filter={test}"'.format(test=test)
-        if clean_env:
-            tmpdir = tempfile.mkdtemp()
-            keys = '{old_key} "--data_path={tmpdir}"'.format(old_key=keys, tmpdir=tmpdir)
-            logging.debug("Temp dir: {tmpdir}".format(tmpdir=tmpdir))
-        else:
-            keys = "{old_key}{resource_path}".format(old_key=keys, resource_path=self.key_postfix)
-            logging.debug("Setting user_resource_path and data_path to {resource_path}".format(resource_path=self.key_postfix))
 
-        self.log_start_finish(test_file, keys, start=True)
-        out, err, result = self.get_tests_from_exec_file(test_file, keys)
-        self.log_start_finish(test_file, keys, finish=True)
+def params_from_dict(params_dict):
+    ret = ""
```

Comment:
return ' '.join(
 	'"{0}={1}"'.format(key, value)
 	for key, value in params_dict.iteritems()
 	if value
) ?